internal/wasm/text/decoder.go

```diff
@@ -91,6 +91,10 @@ type moduleParser struct {
 
 	// unresolvedExports holds any exports whose type index wasn't resolvable when parsed.
 	unresolvedExports map[wasm.Index]*wasm.Export
+
+	// field counts can be different from the count in a section when abbreviated imports exist. To give an accurate
+	// errorContext, we count explicitly.
+	fieldCountFunc, fieldCountMemory uint32
```

Comment:
these will be moved in a later PR to `fieldCount [4]uint32`, so I didn't doc them individually now.